File: day_2/second.php
<?php

$input = file_get_contents('input.txt');
$games = array_filter(explode(PHP_EOL, $input));

$sumOfPowers = 0;
foreach ($games as $game) {
    $minColors = ['red' => 0, 'green' => 0, 'blue' => 0];

    $setList = explode('; ', explode(': ', $game)[1]);
    for ($i = 0; $i < count($setList); $i++) {
        $set = explode(', ', $setList[$i]);
        for ($j = 0; $j < count($set); $j++) {
            list($count, $color) = explode(' ', $set[$j]);
            if ((int)$count > $minColors[$color]) {
                $minColors[$color] = (int)$count;
            }
        }
    }

    $sumOfPowers += $minColors['red'] * $minColors['green'] * $minColors['blue'];
}

echo "Sum of the powers of the minimum sets of cubes: {$sumOfPowers}\n";

File: day_2/second.py
with open('input.txt') as file:
    games = (line.rstrip() for line in file)
    games = list(line for line in games if line)

powerSum = 0
for game in games:
    minColors = {'red' : 0, 'green' : 0, 'blue' : 0}

    sets = game.split(': ')[1].split('; ')
    for set in sets:
        for subSet in set.split(', '):
            count, color = subSet.split(' ')
            if int(count) > minColors[color]:
                minColors[color] = int(count)

    powerSum += minColors['red'] * minColors['green'] * minColors['blue']

print(powerSum)
